### PR TITLE
Vars related cleanup

### PR DESCRIPTION
**Pull request by mkanoor**
Since we have the vars namespace for accessing variables passed in from the vars-file or env_vars we don't have to send the vars to the rule engine. The vars are resolved before we send the ruleset to Drools. Added additional notes in documentation.

### CODE REVIEW

File: docs/conditions.rst [hunk 1]
```diff
@@ -209,7 +209,11 @@ Condition with fact and event
         action:
           debug:
 
-In the above example the custom.expected_index was set using the set_fact action in the running of the rulebook
+| In the above example the custom.expected_index was set using the set_fact action in 
+| the running of the rulebook. You cannot compare a fact and event directly in the same
```
Comment (Alex-Izquierdo) [inline docs/conditions.rst:213]: What about  put some link to here in https://github.com/ansible/ansible-rulebook/blame/main/docs/events_and_facts.rst#L17

Comment (Alex-Izquierdo) [reply docs/conditions.rst:213]: @ttuffin is working also in a new section to detail the usage of variables: https://issues.redhat.com/browse/AAP-8365
pinging him to get aligned

Comment (mkanoor) [reply docs/conditions.rst:213]: @Alex-Izquierdo How do I put links in there so they can be used in readthedocs site. It won't make sense from the readthedocs website to put a link to a raw rst file.

Comment (Alex-Izquierdo) [reply docs/conditions.rst:213]: @mkanoor  here you have an example: https://github.com/ansible/ansible-rulebook/commit/ac5941ccbf8f63b2c68b6ac5a826c62f3184ff88#diff-c49984310d9d7c78bc06bc6bd2e671a6009900ac47d174a66d726056c145ccadR46

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by ttuffin — APPROVED** [review]

**Pull request merged**